File: E_수뭉이와 쿼리/Python/Subtasks_1.py
import sys
input = sys.stdin.readline

def solve(l, r):
    ret = 0
    # L 부터 R 까지 순회
    for i in range(l, r+1):
        if S[i-1] != 'S':
            continue

        # 현재 s[i]가 'S' 이면
        for j in range(i+1, r+1):
            if S[j-1] != 'M':
                continue
            # i 이후의 'M'의 개수를 더함
            ret += 1

    return ret

def run():
    global S
    S, M = input().split()

    M = int(M)
    while M:
        L, R = map(int, input().split())
        print(solve(L, R))
        M -= 1

t = int(input())
while t:
    run()
    t -= 1

File: E_수뭉이와 쿼리/C++/Subtasks_1.cpp
#include <bits/stdc++.h>
using namespace std; using ll = long long;
#define fastio cin.tie(0), cout.tie(0), ios::sync_with_stdio(0);
string S; 

ll solve(ll l, ll r){
    ll ret = 0;

    // L 부터 R 까지 순회
    for(int i = l;i <= r;i++){
        if(S[i - 1] != 'S') continue;

        // 현재 s[i]가 'S' 이면
        for(int j = i + 1;j <= r;j++){
            if(S[j - 1] != 'M') continue;
            // i 이후의 'M'의 개수를 더함
            ret++;
        }
    }

    return ret;
}

void run(){
    ll M;
    cin >> S >> M;
    while(M--){
        ll L, R; cin >> L >> R;
        cout << solve(L, R) << "\n";
    }
}

int main(){
    fastio;
    ll t; cin >> t;
    while(t--) run();
    return 0;
}
